controllers/operands/kubevirtConsolePlugin.go

```diff
@@ -60,28 +60,25 @@ func newKvUIPluginCRHandler(_ log.Logger, Client client.Client, Scheme *runtime.
 func NewKvUIPluginDeplymnt(hc *hcov1beta1.HyperConverged) (*appsv1.Deployment, error) {
 	// The env var was validated prior to handler creation
 	kvUIPluginImage, _ := os.LookupEnv(hcoutil.KVUIPluginImageEnvV)
+	labels := getLabels(hc, hcoutil.AppComponentDeployment)
 
 	return &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      kvUIPluginDeploymentName,
-			Labels:    getLabels(hc, hcoutil.AppComponentDeployment),
+			Labels:    labels,
 			Namespace: hc.Namespace,
 		},
 		Spec: appsv1.DeploymentSpec{
 			Replicas: pointer.Int32(1),
 			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"app": kvUIPluginDeploymentName,
-				},
+				MatchLabels: labels,
```

Comment:
@tiraboschi I looked into this after our chat. The `.spec.selector` and `.spec.template.metadata.labels` fields must be the same, otherwise it will be rejected by the API. [Doc reference](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#selector).

So, I think, we need to consider deleting the Deployment during the upgrade process, as you mentioned during our call.